templates/contest/ranking.html

```diff
@@ -221,6 +221,84 @@
                 {% endif %}
             {% endif %}
         });
+
+        {% if tab == 'ranking' %}
+            $.fn.ignore = function(sel) {
+                return this.clone().find(sel || '>*').remove().end();
+            };
+
+            function download_table_as_csv() {
+                function clean_text(text) {
+                    // Remove new line and leading/trailing spaces
+                    text = text.replace(/(\r\n|\n|\r)/gm, '').trim();
+                    // Escape double-quote with double-double-quote
+                    text = text.replace(/"/g, '""');
+
+                    return text;
+                }
+
+                var csv = [];
+
+                $('table#users-table thead tr').each(function () {
+                    var header = [];
+                    $(this).find('th').each(function () {
+                        var $col = $(this);
+
+                        if ($col.hasClass('rating-column')) {
+                            // Skip rating
+                            return;
+                        } else if ($col.hasClass('personal-info')) {
+                            // Full name
+                            header.push('Full name');
+                        } else if ($col.hasClass('username') || $col.hasClass('rank')) {
+                            // Username or rank
+                            header.push(clean_text($col.text()));
+                        } else {
+                            // Point
+                            var name = $col.find('.problem-code').text();
+                            if (name == '') {
+                                name = $col.text();
+                            }
+                            header.push(clean_text(name));
+                        }
+                    });
+                    csv.push(header.join(','));
+                });
+
+                $('table#users-table tbody tr').each(function () {
+                    var row_data = [];
+                    $(this).find('td').each(function () {
+                        var $col = $(this);
+
+                        if ($col.hasClass('rating-column')) {
+                            // Skip rating
+                            return;
+                        } else if ($col.hasClass('personal-info')) {
+                            // Full name
+                            row_data.push(clean_text($col.children().first().text()));
+                        } else if ($col.hasClass('user-name')) {
+                            // Username
+                            row_data.push(clean_text($col.text()));
+                        } else {
+                            // Point or rank
+                            row_data.push(clean_text($col.ignore('.solving-time').text()));
+                        }
+                    });
+                    csv.push(row_data.join(','));
```

Comment:
since the user's full name can has `,` in it, I think we should escape here.

Also, note that full name can has `"` as well